File: src/main/java/ejercicioJava/SistemaGestionEstudiantes/Profesor.java
package ejercicioJava.SistemaGestionEstudiantes;

import java.util.List;

public class Profesor {
    public String nombre;

    public Profesor(String nombre) {
        this.nombre = nombre;
    }

    public boolean asignarCurso(Estudiante estudiante, Curso curso) {
        List<Curso> cursos = estudiante.getCursos();
        if (!cursos.contains(curso)) {
            cursos.add(curso);
            return true;
        }
        return false;
    }

    public int calificarEstudiante(Estudiante estudiante) {
        int notaMedia = 0;
        List<Curso> cursos = estudiante.getCursos();
        for (Curso curso : cursos) {
            notaMedia += curso.getNotaAlumno();
        }
        return notaMedia / cursos.size();
    }
}

File: src/main/kotlin/ejercicioKotlin/sistemaGestionEstudiantes/Profesor.kt
package org.example.ejercicioKotlin.sistemaGestionEstudiantes

class Profesor(val nombre:String) {

    fun asignarCurso(estudiante: Estudiante, curso: Curso):Boolean {
        if (curso !in estudiante.cursos) {
            estudiante.cursos.add(curso)
            return true
        }
        return false

    }

    fun calificarEstudiante(estudiante: Estudiante):Int {
        var notaMedia = 0
        estudiante.cursos.forEach { notaMedia += it.notaALumno }
        return notaMedia / estudiante.cursos.size
    }

}